// ng-marquee-improved/bundles/ng-marquee-improved.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ng-marquee-improved', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global['ng-marquee-improved'] = {}, global.ng.core));
}(this, function (exports, core) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgMarqueeComponent = /** @class */ (function () {
        function NgMarqueeComponent() {
        }
        /**
         * @return {?}
         */
        NgMarqueeComponent.prototype.updateDuration = /**
         * @return {?}
         */
        function () {
            if (!this.duration) {
                return;
            }
            this.animationElem.style.animationDuration = this.duration + "s";
        };
        /**
         * @return {?}
         */
        NgMarqueeComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () { };
        /**
         * @return {?}
         */
        NgMarqueeComponent.prototype.ngAfterViewInit = /**
         * @return {?}
         */
        function () {
            this.animationElem = (/** @type {?} */ (document.querySelector(".ng-marquee-wrapper#" + this.marqueeId + " > div")));
            this.updateDuration();
        };
        NgMarqueeComponent.decorators = [
            { type: core.Component, args: [{
                        selector: "ng-marquee",
                        template: "<div class=\"ng-marquee-wrapper\" id=\"{{ marqueeId }}\">\n  <div>\n    <ng-content></ng-content>\n  </div>\n</div>\n",
                        changeDetection: core.ChangeDetectionStrategy.OnPush,
                        styles: [".ng-marquee-wrapper{overflow:hidden;position:relative}.ng-marquee-wrapper>div{height:100%;display:inline-block;margin:0;padding-left:100%;-webkit-animation:20s linear infinite marqueeAnimation;animation:20s linear infinite marqueeAnimation;white-space:nowrap}.ng-marquee-wrapper:hover>div{-webkit-animation-play-state:paused;animation-play-state:paused}@-webkit-keyframes marqueeAnimation{0%{-webkit-transform:translateX(0);transform:translateX(0)}100%{-webkit-transform:translateX(-100%);transform:translateX(-100%)}}@keyframes marqueeAnimation{0%{-webkit-transform:translateX(0);transform:translateX(0)}100%{-webkit-transform:translateX(-100%);transform:translateX(-100%)}}@media (max-width:500px){.ng-marquee-wrapper>div{width:100%;height:100%;margin:0;-webkit-transform:translateX(100%);transform:translateX(100%);-webkit-animation:9s linear 1s infinite marqueeAnimation;animation:9s linear 1s infinite marqueeAnimation;white-space:nowrap}}"]
                    }] }
        ];
        /** @nocollapse */
        NgMarqueeComponent.ctorParameters = function () { return []; };
        NgMarqueeComponent.propDecorators = {
            duration: [{ type: core.Input }],
            marqueeId: [{ type: core.Input }]
        };
        return NgMarqueeComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgMarqueeModule = /** @class */ (function () {
        function NgMarqueeModule() {
        }
        NgMarqueeModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [],
                        declarations: [NgMarqueeComponent],
                        exports: [NgMarqueeComponent]
                    },] }
        ];
        return NgMarqueeModule;
    }());

    exports.NgMarqueeComponent = NgMarqueeComponent;
    exports.NgMarqueeModule = NgMarqueeModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
